monetio/sat/_tropomi_l2_no2_mm.py

```diff
@@ -103,9 +103,8 @@ def get_extra(varname_, *, dct_=None, default_group="PRODUCT"):
             # Tropopause pressure
             ptrop = xr.full_like(itrop, np.nan, dtype=ds["preslev"].dtype)
             for i in np.unique(itrop):
-                if np.isnan(i):
-                    continue
-                ptrop = xr.where(itrop == i, p.isel(z=int(i)), ptrop)
+                if i >= 0 and i < p.sizes["z"]:
```

Comment:
When would `i` be outside this range?

Reply:
In the OFFL files there are cases where the "tm5_tropopause_layer_index" has values outside of this range, which causes a problem when those values are used to index into the pressure array to get the tropopause pressure.

Reply:
Just  $\ge$ `p.sizes["z"]`? or negative values too?

Reply:
There were some negative values included.